presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -995,11 +996,11 @@ private String validateAndNormalizeAvroSchemaUrl(String url, HdfsContext context
         }
     }
 
-    private Path getExternalPath(HdfsContext context, String location)
+    private Path getExternalPath(HdfsContext context, String location, boolean isNewTable)
```

Comment:
It's always used in case of "new table", but the data is expected or not to exists, so `isNewData` would be appreopriate.

However, this method is just a check, it's not supposed to return anything other than `new Path(location)`.
So I propose

- in case of CTAS, do not call it
- in case of CT, call it, but change the method signature: -> `checkExternalPath(HdfsContext, Path)`